lib/view_component/base.rb

```diff
@@ -108,11 +111,13 @@ def render(options = {}, args = {}, &block)
     end
 
     def controller
+      raise ViewContextCalledBeforeRenderError, "Cannot call `controller` from initialize. Must be called during render" unless @render_in_called
```

Comment:
Two things:

1) Do you think we could simplify this check to see if `view_context` is defined or not, instead of using `@render_in_called`? 
2) I wonder if we could be more specific with the error message:

```suggestion
      raise ViewContextCalledBeforeRenderError, "`controller` can only be called at render time.
```

Reply:
> Do you think we could simplify this check to see if view_context is defined or not, instead of using @render_in_called?

I considered this approach, but the ivar seemed more direct instead of implicit. I'm cool either way, WDYT?

>  wonder if we could be more specific with the error message:

Totally!